### PR TITLE
support acl command

### LINKED ISSUE #35: [API Proposal] ACL Commands
#### Syntax
```
 ACL CAT [category]
```
The command shows the available ACL categories if called without arguments. If a category name is given, the command shows all the Quebrix commands in the specified category.

#### Syntax
```
    ACL DELUSER username [username ...]
```
Delete all the specified ACL users and terminate all the connections that are authenticated with such users. Note: the special default user cannot be removed from the system, this is the default user that every new connection is authenticated with. The list of users may include usernames that do not exist, in such case no operation is performed for the non existing users.

#### Syntax
```
    ACL LIST
```
The command shows the currently active ACL rules in the Quebrix server.

#### Syntax
```
    ACL LOAD
```
When Quebrix is configured to use an ACL file, this command will reload the ACLs from the file, replacing all the current ACL rules with the ones defined in the file. The command makes sure to have an all or nothing behavior, that is:

- If every line in the file is valid, all the ACLs are loaded.
- If one or more line in the file is not valid, nothing is loaded, and the old ACL rules defined in the server memory continue to be used.

#### Syntax
```
    ACL SETUSER username [rule [rule ...]]
```
Create an ACL user with the specified rules or modify the rules of an existing user.

Manipulate Quebrix ACL users interactively. If the username does not exist, the command creates the username without any privilege. It then reads from left to right all the rules provided as successive arguments, setting the user ACL rules as specified. If the user already exists, the provided ACL rules are simply applied in addition to the rules already set.

#### Syntax
```
    ACL USERS
```
The command shows a list of all the usernames of the currently configured users in the Quebrix ACL system.

#### Syntax
```
    ACL WHOAMI
```
Return the username the current connection is authenticated with. New connections are authenticated with the "default" user. They can change user using AUTH.

### PR DESCRIPTION
**Pull request by SaeedSafi1999**
clean up cred manager code 
add some apis
delete 
set 
load users

### CODE REVIEW

File: src/creds/auth.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                        return true;
+                    } else {
+                        if self.enable_log == true {
+                            let message =
+                                &format!("password not match with user : {:?}", username.clone());
+                            let set_log = Logger::log_warn_data(message);
+                            set_log.write_log_to_file();
+                        }
+                        println!("Passwords do not match for user: {}", username);
+                        return false;
```
Comment (Alirexaa) [inline src/creds/auth.rs:26]: this is dupe code in this file, please make a function for this.

File: src/creds/cred_manager.rs [hunk 2]
```diff
@@ -63,12 +89,12 @@ impl std::fmt::Display for ParseUserRoleError {
...
-    users: HashMap<String, User>,
-    enable_log: bool,
+    pub users: HashMap<String, User>,
+    pub enable_log: bool,
 }
 
-const ENCRYPT_KEY: &str = "QWERTYUIOPASdfgCBxzjfndeuAQwudhAsd";
-const IV_PATTERN: [u8; 16] = [2, 2, 5, 8, 5, 1, 3, 5, 6, 7, 9, 1, 6, 4, 1, 9];
+pub const ENCRYPT_KEY: &str = "QWERTYUIOPASdfgCBxzjfndeuAQwudhAsd";
+pub const IV_PATTERN: [u8; 16] = [2, 2, 5, 8, 5, 1, 3, 5, 6, 7, 9, 1, 6, 4, 1, 9];
```
Comment (Alirexaa) [inline src/creds/cred_manager.rs:97]: we should'nt hard code `ENCRYPT_KEY` and `IV_PATTERN`. we should have these in the configuration file. It should be optional.
Some random `ENCRYPT_KEY` and `IV_PATTERN` should be created if not provided.

File: src/creds/acl_add_user.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+use crate::{crypto::crypto_service::Encryptor, logger::logger_manager::Logger};
+
+use super::cred_manager::{ACLResult, CredsManager, RoleManagement, User, ENCRYPT_KEY, IV_PATTERN};
+
+pub trait IAddUser {
```
Comment (Alirexaa) [inline src/creds/acl_add_user.rs:5]: Do we need `I` prefix for our traits?

File: src/creds/del_user.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        if self.users.remove(username).is_none() {
+            return ACLResult::faild("User not found");
+        }
+
+        let remove_result = self.remove_user_from_file(username);
+        if !remove_result.is_success {
+            return ACLResult::faild("can not remove from file");
+        }
```
Comment (Alirexaa) [inline src/creds/del_user.rs:34]: These operations should be atomic. The user will delete from the file and `self.users` or the one that never gets deleted.

File: src/creds/auth.rs (hunk 1 above)
Comment (SaeedSafi1999) [reply src/creds/auth.rs:26]: its better to do this later we have lot of work  to do now for the project.
I will create a new issue for cleanup code

File: src/creds/cred_manager.rs (hunk 2 above)
Comment (SaeedSafi1999) [reply src/creds/cred_manager.rs:97]: no it must do by product owner not in configuration cause when we want to sell this product to BS we need handle by our selves, not client .in open source version they go to code and change it

File: src/creds/acl_add_user.rs (hunk 3 above)
Comment (SaeedSafi1999) [reply src/creds/acl_add_user.rs:5]: its global contracts with other programmers like interface in other language it cause more readable for other developers.

File: src/creds/del_user.rs (hunk 4 above)
Comment (SaeedSafi1999) [reply src/creds/del_user.rs:34]: this atomic now!!!

Comment (SaeedSafi1999) [reply src/creds/del_user.rs:34]: it also delete from internal app hashmap user and in file in application

**Comment by SaeedSafi1999** [comment]
CC @Alirexaa merge it we do some stuff later about clean up code

File: src/creds/auth.rs (hunk 1 above)
Comment (SaeedSafi1999) [reply src/creds/auth.rs:26]: it is so major!!!

Comment (SaeedSafi1999) [reply src/creds/auth.rs:26]: i made issue for this we can handle it later
in issue #44

File: src/creds/cred_manager.rs (hunk 2 above)
Comment (SaeedSafi1999) [reply src/creds/cred_manager.rs:97]: I made issue #45 for this

**Review by Alirexaa — APPROVED** [review]

**Pull request merged**